metabase/metabase
# Allowlist for temporal unit parameters

Epic https://github.com/metabase/metabase/issues/42118

Adds a parameter setting to specify allowed temporal units for selection.

Demo: https://www.loom.com/share/3e1329e459454b8d947d9af66b6d2b0d

How to verify:
- Add a question with a breakout on a dashboard
- Add a temporal unit parameter and connect it to the question
- There is new "Unit of Time options" parameter setting available in the sidebar
- Change the list of units and save the dashboard
- Make sure that only the selected units are available in the widget

Makes `SelectItem` available as a standalone component.

<img width="412" alt="Screenshot 2024-06-10 at 15 56 06" src="https://github.com/metabase/metabase/assets/8542534/4dba6ccd-bb52-45c8-806f-a75c8f0e918e">
<img width="1185" alt="Screenshot 2024-06-10 at 15 56 12" src="https://github.com/metabase/metabase/assets/8542534/42b4f8d8-9264-4f86-8325-c646e70ba27f">
